python/src/aiconfig/editor/client/src/components/ParametersRenderer.tsx

```diff
@@ -0,0 +1,253 @@
+import {
+  Group,
+  Text,
+  TextInput,
+  Textarea,
+  ActionIcon,
+  Stack,
+  useMantineTheme,
+  Tooltip,
+} from "@mantine/core";
+import { IconTrash, IconPlus } from "@tabler/icons-react";
+import { debounce, uniqueId } from "lodash";
+import { useState, useCallback, memo, useMemo } from "react";
+
+interface JSONArray extends Array<JSONValue> {}
+
+interface JSONObject {
+  [x: string]: JSONValue;
+}
+
+type JSONValue = string | number | boolean | JSONObject | JSONArray | unknown;
+
+type Parameter = { parameterName: string; parameterValue: JSONValue };
+
+/**
+ * Parameter name must start with a letter (a-z, A-Z) or an underscore (_). The rest of the
+ * name can contain letters, digits (0-9), underscores, and dollar signs ($).
+ */
+export function isValidParameterName(name: string): boolean {
+  const validNamePattern = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+  return validNamePattern.test(name);
+}
+
+const ParameterInput = memo(function ParameterInput(props: {
+  onUpdateParameter: (data: {
+    promptName?: string;
+    parameterName: string;
+    oldParameterName?: string;
+    parameterValue?: string;
+  }) => void;
+  initialItemValue?: Parameter;
+  removeParameter: (parameterName?: string) => Promise<void>;
+}) {
+  const { initialItemValue, removeParameter, onUpdateParameter } = props;
+  // TODO: saqadri - update this once we have a readonly mode
+  const { isReadonly } = { isReadonly: false };
+
+  const [parameterName, setParameterName] = useState(
+    initialItemValue?.parameterName ?? ""
+  );
+  const [lastParameterName, setLastParameterName] =
+    useState<string>(parameterName);
+
+  const parameterValue = initialItemValue?.parameterValue;
+
+  const [parameterValueString, setParameterValueString] = useState(
+    typeof parameterValue === "string"
+      ? parameterValue
+      : JSON.stringify(parameterValue)
+  );
+
+  const debouncedCellParameterUpdate = useMemo(
+    () =>
+      debounce((newParameterName: string, newParameterValue: string) => {
+        if (!isValidParameterName(newParameterName)) {
+          return;
+        }
+
+        onUpdateParameter({
+          oldParameterName: lastParameterName,
+          parameterName: newParameterName,
+          parameterValue: newParameterValue,
+        });
+
+        setLastParameterName(newParameterName);
+      }, 250),
+    [lastParameterName, onUpdateParameter]
+  );
+
+  const theme = useMantineTheme();
```

Comment:
Just a note that we'll need to add the MantineProvider wrapper inside EditorContainer in order for this to pick up context; that's something we can do later with our styling/UI cleanup pass

Reply:
Added a TODO comment stating as such